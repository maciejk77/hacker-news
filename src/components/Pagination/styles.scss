@import '../../styles/base_styles.scss';

.pagination {
  margin-bottom: 2rem;
  padding: 0.5rem;
  font-family: $fonts;
  font-size: $font-size-sm;

  .pagination-list {
    display: flex;
    justify-content: center;

    &__item {
      background-color: $bg-col1;
      margin: 0 0.2rem;
      padding: 0.5rem;

      .item {
        color: $txt-col1;
        text-decoration: none;
      }
    }
  }
}
