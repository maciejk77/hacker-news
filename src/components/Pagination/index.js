import React from 'react';
import './styles.scss';

const Pagination = ({ storiesPerPage, totalStories, paginate }) => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalStories / storiesPerPage); i++) {
    pageNumbers.push(i);
  }

  return (
    <nav className="pagination">
      <ul className="pagination-list">
        {pageNumbers.map(number => (
          <li key={number} className="pagination-list__item">
            <a onClick={() => paginate(number)} href="!#">
              {number}
            </a>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Pagination;
