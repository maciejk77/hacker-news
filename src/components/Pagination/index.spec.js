import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../Pagination';

describe('<Pagination />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });
});
