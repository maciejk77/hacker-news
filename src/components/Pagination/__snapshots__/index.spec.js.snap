// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `ShallowWrapper {}`;
